rdkPlugins/Logging/source/LoggingPlugin.cpp

```diff
@@ -274,70 +268,45 @@ void LoggingPlugin::JournaldSink(const ContainerInfo &containerInfo, bool exitEo
         }
     }
 
-    // Read from the fd until the file is closed
-    // Journald expects messages to be single lines, so we need to process incoming
-    // data to split into individual lines
+    char buf[PTY_BUFFER_SIZE];
+    memset(buf, 0, sizeof(buf));
+
+    ssize_t ret;
+    ssize_t offset = 0;
+
+    // Create a file descriptor we can write to
+    // journald will handle line breaks etc automatically so we don't need
+    // to worry about them
+    int outputFd = sd_journal_stream_fd(mUtils->getContainerId().c_str(), logPriority, 1);
+
     while (!cancellationToken)
     {
-        // Work out how much space is left in the buffer and read as much as we can
-        bufferRemaining = sizeof(buf) - bufferUsed;
-
-        bytesRead = read(containerInfo.pttyFd, &buf[bufferUsed], bufferRemaining);
-        if (bytesRead < 0)
+        ret = read(containerInfo.pttyFd, buf, sizeof(buf));
+        if (ret < 0)
         {
             AI_LOG_INFO("Container %s terminated, terminating logging thread",
                         mUtils->getContainerId().c_str());
             break;
         }
-        if (bytesRead == 0 && exitEof)
+        if (ret == 0 && exitEof)
         {
             break;
         }
-        bufferUsed += bytesRead;
-
-        // Loop through the received data, finding the position of new lines. Only
-        // send complete lines to journald
-        char *lineStart = buf;
-        char *lineEnd;
-        while ((lineEnd = static_cast<char *>(memchr(lineStart, '\n', bufferUsed - (lineStart - buf)))) != nullptr)
-        {
-            *lineEnd = 0;
-
-            std::string msg(lineStart, strlen(lineStart));
-
-            // Despite what documentation says, we need to remove any line break
-            // characters. Only matched on the \n so we still need to remove the
-            // \r if there is one
-            if (msg.back() == '\r')
-            {
-                msg.pop_back();
-            }
-
-            if (!msg.empty())
-            {
-                // Note PID in "journalctl -f" will show as daemon PID, but when
-                // viewing journald in full JSON format, the container PID is
-                // visible
-                sd_journal_send("MESSAGE=%s", msg.c_str(),
-                                "PRIORITY=%i", logPriority,
-                                "SYSLOG_IDENTIFIER=%s", mUtils->getContainerId().c_str(),
-                                "OBJECT_PID=%ld", containerInfo.containerPid,
-                                "SYSLOG_PID=%ld", containerInfo.containerPid,
-                                NULL);
-            }
-
-            lineStart = lineEnd + 1;
-        }
+        offset += ret;
```

Comment:
This offset variable seams unused. What should it do?

Reply:
not needed anymore, leftover from old implementation. Removed